tests/Clarifai/Repository/SearchInputRepositoryTest.php

```diff
@@ -0,0 +1,354 @@
+            ->with(
+                'POST',
+                'searches',
+                [
+                    'query' => [
+                        'ands' => [
+                            [
+                                'output' => [
+                                    'data' => [
+                                        'concepts' => [
+                                            [
+                                                'name' => $concept->getName(),
+                                                'value' => $concept->getValue(),
+                                            ],
+                                        ],
+                                    ],
+                                ],
+                            ],
+                        ],
+                    ],
+                ]
+            )
+            ->andReturn($this->generateInputSearchResult([$input->setConcepts([$concept])]));
+
+        $this->assertEquals(
+            [$input],
+            $this->searchInputRepository->searchByPredictedConcepts([$concept])
+        );
+    }
+
+    public function testSearchByUserSuppliedConcepts()
+    {
+        $input1 = $this->getFullInputEntity();
+        $input2 = $this->getFullInputEntity()->setId('id2')->setImage('image2');
+
+        $concept = $this->getFullConceptEntity('id1', true);
+
+        $this->request->shouldReceive('request')
+            ->once()
+            ->with(
+                'POST',
+                'searches',
+                [
+                    'query' => [
+                        'ands' => [
+                            [
+                                'input' => [
+                                    'data' => [
+                                        'concepts' => [
+                                            [
+                                                'name' => $concept->getName(),
+                                                'value' => $concept->getValue(),
+                                            ],
+                                        ],
+                                    ],
+                                ],
+                            ],
+                        ],
+                    ],
+                ]
+            )
+            ->andReturn(
+                $this->generateInputSearchResult(
+                    [
+                        $input1->setConcepts([$concept]),
+                        $input2->setConcepts([$concept]),
+                    ]
+                )
+            );
+
+        $this->assertEquals(
+            [$input1, $input2],
+            $this->searchInputRepository->searchByUserSuppliedConcepts([$concept])
+        );
+    }
+
+    public function testSearchByReverseImage()
+    {
+        $input1 = $this->getFullInputEntity()->setId('id1')->setImage('image');
+        $input2 = $this->getFullInputEntity()->setId('id2')->setImage('image');
+
+        $this->request->shouldReceive('request')
```

Comment:
Also here you have a lot of almost the same arrays, just with one or two different keys. Consider to refactor them and move to helper method.